src/teacher/StudentReadingInsights.js

```diff
@@ -14,7 +15,22 @@ export default function StudentReadingInsights({ api }) {
   const [studentName, setStudentName] = useState("");
   const [cohortLang, setCohortLang] = useState("");
   const [readArticles, setReadArticles] = useState([]);
-  const [articleCount, setArticleCount] = useState(0);
+  const [articleCount, setArticleCount] = useState(-1);
+
+  useEffect(() => {
+    api.getCohortsInfo((cohortInfo) => {
+      let currentCohort = cohortInfo.filter((each) => each.id === cohortID);
```

Comment:
something like this would read nicer ;)

```      
let currentCohort = cohortInfo.find((each) => each.id === cohortID);
setCohortLang(currentCohort.language_name);
``` 

and would have the same behavior

Reply:
Fixed